### PR TITLE
Updates according to feedbacks from last pull request

### PR DESCRIPTION
**Pull request by nowshin-108**
1. Removed frontend authentication verification and managed all of it in backend.
2. Redirecting to login if authentication fails
3. Removed session storage for add trip and now passing useContext to manage state
4. Created get user data route to retrieve usr data for later

### CODE REVIEW

File: backend/server.js [hunk 1]
```diff
@@ -23,9 +23,9 @@ app.use(
   session({
     secret: 'your-secret-key',
     resave: false,
-    saveUninitialized: false,
+    saveUninitialized: true,
```
Comment (alantulais) [inline backend/server.js:26]: What's the purpose of this change?

File: frontend/src/App.jsx [hunk 2]
```diff
@@ -14,48 +14,63 @@ import { API_BASE_URL } from './config.js';
...
+    setIsLoading(true);
+    try {
+      const response = await fetch(`${API_BASE_URL}/users/data`, {
+        credentials: 'include',
+      });
+      if (response.ok) {
+        const userData = await response.json();
+        updateUser(userData);
+      } else {
+        setUser(null);
```
Comment (alantulais) [inline frontend/src/App.jsx:31]: Shall we throw an exception here to use same logic as in catch block?

File: frontend/src/Components/HomePage/AddTrip/FlightForm.jsx [hunk 3]
```diff
@@ -50,31 +29,31 @@ const FlightForm= () => {
...
             
             if (response.data && response.data.data && response.data.data.length > 0) {
-                setFlightStatus(response.data);
+                updateFlightData({ flightStatus: response.data });
             } else {
                 setError("No flights found. Please check your input and try again.");
             }
         
         } catch (err) {
             if (err.response && err.response.status === 401) {
```
Comment (alantulais) [inline frontend/src/Components/HomePage/AddTrip/FlightForm.jsx:54]: I would suggest to create a helper function to handle this, so that you can reuse this check every time you catch an API exception.

File: frontend/src/App.jsx [hunk 4]
```diff
@@ -14,48 +14,63 @@ import { API_BASE_URL } from './config.js';
...
   
   return (
     <div className="app">
-      <UserContext.Provider value={{ user, updateUser }}>
+      <UserContext.Provider value={{ user, updateUser, flightData, updateFlightData, fetchUserData }}>
         <LoadingProvider>
           <BrowserRouter>
             <Routes>
-              <Route path="/" element={user ? <HomePage/>: <LoginForm />} />
+              {isLoading ? (
```
Comment (abeadam) [inline frontend/src/App.jsx:60]: instead of doing this. How about creating a specific page with a loader if we want to go that route and redirect to it https://reactrouter.com/en/main/fetch/redirect . Alternatively showing a loader component on the active page would look nicer.

File: frontend/src/Components/HomePage/AddTrip/FlightForm.jsx [hunk 5]
```diff
@@ -109,15 +88,14 @@ const FlightForm= () => {
             return;
         }
         
-        if (!flightStatus || !flightStatus.data || flightStatus.data.length === 0) {
+        if (!flightData === 0) {
```
Comment (abeadam) [inline frontend/src/Components/HomePage/AddTrip/FlightForm.jsx:91]: `flightData !== 0` though I would prefer seeing it to null if it doesn't exist and using numerical check if you're looking at the length of an array

File: frontend/src/Components/LoginForm/LoginForm.jsx [hunk 6]
```diff
@@ -30,6 +30,7 @@ const LoginForm = () => {
       if (response.ok) {
         const data = await response.json();
         updateUser(data.user);
+        await fetchUserData();
```
Comment (abeadam) [inline frontend/src/Components/LoginForm/LoginForm.jsx:33]: if you don't need to wait on the data request at line 31, you can send out two requests at the same time with `Promise.all`

**Review by alantulais — CHANGES_REQUESTED** [review]
@nowshin-108 Are you planning to update this PR or is it ready to approve?
